### PR TITLE
IdentityHubCredentialsVerifier fails if credential is invalid

### PR DESCRIPTION
**Pull request by ouphi**
## What this PR changes/adds

Changed the logic of the IdentityhubCredentialsVerifier to fail if one credential has an invalid format.
The long term goal is that the CredentialsVerifier returns Failures and success and give the choice to the client to succeed or fail.

Next steps: Will do a second PR on the same feature branch to fail if one credential has invalid signature or signature can't be verified.

## Linked Issue(s)

https://github.com/agera-edc/IdentityHub/issues/47

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/identity-hub-verifier/src/test/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifierTest.java [hunk 1]
```diff
@@ -74,9 +73,6 @@ public void getVerifiedClaims_getValidClaims() throws Exception {
 
         // Assert
         assertThat(credentials.succeeded()).isTrue();
-        assertThat(credentials.getContent())
```
Comment (algattik) [inline extensions/identity-hub-verifier/src/test/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifierTest.java]: why is this removed?

**Review by algattik — CHANGES_REQUESTED** [review]

File: extensions/identity-hub-verifier/src/test/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifierTest.java (hunk 1 above)
Comment (ouphi) [reply extensions/identity-hub-verifier/src/test/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifierTest.java]: Ah my mistake, reverted it.

File: extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java [hunk 2]
```diff
@@ -92,7 +93,12 @@ public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocumen
 
         var claims = extractClaimsFromCredential(verifiedCredentials);
 
-        return Result.success(claims);
+        if (claims.failed()) {
+            monitor.severe(String.format("Credentials verification failed: %s", claims.getFailureDetail()));
```
Comment (algattik) [inline extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java]:
```suggestion
            monitor.severe(() -> String.format("Credentials verification failed: %s", claims.getFailureDetail()));
```
best practice to avoid computing string if logging disabled at that level

**Review by algattik — APPROVED** [review]

File: extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java [hunk 3]
```diff
@@ -138,4 +143,10 @@ private String getIdentityHubBaseUrl(DidDocument didDocument) {
                 .map(Service::getServiceEndpoint)
                 .orElse(null);
     }
+
+    private static class VerificationResult extends AbstractResult<Map<String, Object>, Failure> {
```
Comment (marcgs) [inline extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java:147]: The namings seem a bit off: the method is called `extractClaimsFromCredential` but the result is a `VerificationResult` which gets assigned to a variable called `claims`. What about renaming this class to `ClaimsExtractionResult`?

**Review by marcgs — APPROVED** [review]

File: extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java (hunk 3 above)
Comment (ouphi) [reply extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java:147]: Using the Result type and calling the variable by what the Result contains is not so odd, and it is done in other places.

I would also move this discussion to the next PR, because I make this class more generic in the next PR, to be able to return Failures and success when verifying SignedJWT (so it will not be only specific to claims).

Comment (marcgs) [reply extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java:147]: Ok agreed, lets discuss in the next PR
